src/app/services/project_service.js

```diff
@@ -716,6 +719,53 @@ angular
         return databases;
     }
 
+
+    function buildGroupTree(groups, nodes, select) {
+        var groups = {}
+
+        _.each(nodes, function(node) {
+            var show = _.get(node, ['docs', 'show'], true);
+            var exclude_nodes = ['source', 'exposure', 'seed', 'macro']
+            if (node.resource_type in exclude_nodes || !show || node.access === "private") {
+                return;
+            }
+
+            var name = node.name;
+
+            var group = node.group;
+
+            var is_active = node.unique_id == select;
+
+            if (!groups[group]) {
+                groups[group] = {
+                    type: "group",
+                    name: group,
+                    active: is_active,
+                    items: []
+                };
+            } else if (is_active) {
+                groups[group].active = true;
+            }
+
+            groups[group].items.push({
+                type: 'file',
+                name: name,
+                node: node,
+                active: is_active,
+                unique_id: node.unique_id,
+                node_type: 'group'
+            })
+        });
+
+        var groups = _.sortBy(_.values(groups), 'name');
```

Comment:
Since `groups` is an object, sorting it seems a bit odd since objects are likened to dicts in python. If you wanted a sorted list of keys from `groups`, I'd do something like `Object.keys(groups).sort()` then use this sorted array of keys to access `groups`.

Reply:
This sorts the groups by the name value.